extensions/notebook/src/dialog/remoteBookDialog.ts

```diff
@@ -0,0 +1,249 @@
+			this.urlInputBox.focus();
+		});
+		let downloadButton = azdata.window.createButton('Download');
+		downloadButton.onClick(async () => await this.download());
+		this.dialog.customButtons = [];
+		this.dialog.customButtons.push(downloadButton);
+		this.dialog.okButton.onClick(async () => await this.openRemoteBook());
+		this.dialog.okButton.label = loc.open;
+		this.dialog.okButton.enabled = false;
+		this.dialog.cancelButton.label = loc.cancel;
+		azdata.window.openDialog(this.dialog);
+	}
+
+	private get remoteLocationValue(): string {
+		return (<azdata.CategoryValue>this.remoteLocationDropdown.value).name;
+	}
+
+	private get remoteBookValue(): string {
+		return this.remoteBookDropdown.value.toString();
+	}
+
+	private onRemoteLocationChanged(): void {
+		let location = this.remoteLocationValue;
+		if (this.releases !== undefined && location === 'GitHub') {
+			this.releaseDropdown.updateCssStyles({
+				display: 'block'
+			});
+		} else {
+			this.releaseDropdown.updateCssStyles({
+				display: 'none'
+			});
+		}
+	}
+
+	private async validate(): Promise<void> {
+		let url = this.urlInputBox && this.urlInputBox.value;
+		let location = this.remoteLocationValue;
+
+		try {
+			if (location === 'GitHub') {
+				let releases = await this.model.getReleases(url);
+				if (releases !== undefined && releases !== []) {
+					this.releaseDropdown.updateCssStyles({
+						display: 'block'
+					});
+					await this.fillReleasesDropdown(releases);
+				}
+			}
+		}
+		catch (error) {
+			this.dialog.message = {
+				text: (typeof error === 'string') ? error : error.message,
+				level: azdata.window.MessageLevel.Error
+			};
+		}
+	}
+
+	private async download(): Promise<void> {
+		let location = this.remoteLocationValue;
+		this.dialog.customButtons[0].enabled = false;
+		let books: string[] = [];
+		try {
+			if (location === 'GitHub') {
+				let selected_release = this.releases.filter(release =>
+					release.tag_name === this.releaseDropdown.value);
+				books = await this.model.downloadLocalCopy(selected_release[0].remote_path, location, selected_release[0]);
+			} else {
+				let url = this.urlInputBox && this.urlInputBox.value;
+				let newUrl = new URL(url);
+				books = await this.model.downloadLocalCopy(newUrl, location);
+			}
+			if (books !== undefined && books.length > 0) {
+				this.remoteBookDropdown.updateCssStyles({
+					display: 'block'
+				});
+				this.remoteBookDropdown.updateProperties({
+					values: books
+				});
+				this.dialog.okButton.enabled = true;
+				this.dialog.message = {
+					text: 'No Books available to open',
+					level: azdata.window.MessageLevel.Information
+				};
+			}
+		}
+		catch (error) {
+			this.dialog.message = {
+				text: (typeof error === 'string') ? error : error.message,
+				level: azdata.window.MessageLevel.Error
+			};
+		}
+	}
+
+	private async fillReleasesDropdown(releases: IReleases[]): Promise<void> {
+		this.releases = releases;
+		let versions: string[] = [];
```

Comment:
You can shorten this to `const versions: string[] = releases.map(release => release.tag_name);`